src/components/DateInput.vue

```diff
@@ -113,30 +113,29 @@ export default {
         13  // enter
       ].includes(event.keyCode)) {
         this.input.blur()
-        this.inputBlurred()
-        this.$emit('closeCalendar')
-        return false
       }
 
-      const typedDate = Date.parse(this.input.value)
-      if (!isNaN(typedDate)) {
-        this.typedDate = this.input.value
-        this.$emit('typedDate', new Date(this.typedDate))
+      if (this.typeable) {
+        const typedDate = Date.parse(this.input.value)
+        if (!isNaN(typedDate)) {
+          this.typedDate = this.input.value
+          this.$emit('typedDate', new Date(this.typedDate))
+        }
       }
     },
     /**
      * nullify the typed date to defer to regular formatting
      * called once the input is blurred
      */
     inputBlurred () {
-      if (!this.typedDate) {
-        return
-      }
-      if (isNaN(Date.parse(this.input.value))) {
-        this.clearDate()
+      if (this.typedDate && this.typeable) {
```

Comment:
I'd prefer to keep the nesting level lower and return if we don't pass this.
I find this style much easier to read and reason about.

Additionally `typedDate` should never be truthy if `typeable` is false.

Reply:
Will fix.